siriuspy/siriuspy/devices/currinfo.py

```diff
@@ -1,158 +1,222 @@
 """."""
 from .device import Device as _Device
+from .device import Devices as _Devices
 
 
-class CurrInfo(_Device):
+class CurrInfoLinear(_Device):
     """."""
 
     class DEVICES:
         """Devices names."""
 
         LI = 'LI-Glob:AP-CurrInfo'
         TB = 'TB-Glob:AP-CurrInfo'
-        BO = 'BO-Glob:AP-CurrInfo'
         TS = 'TS-Glob:AP-CurrInfo'
-        SI = 'SI-Glob:AP-CurrInfo'
+        ALL = (LI, TB, TS, )
+
+    _properties = ('TranspEff-Mon', )
+
+    def __init__(self, devname):
+        """."""
+        # check if device exists
+        if devname not in CurrInfoLinear.DEVICES.ALL:
+            raise NotImplementedError(devname)
+
+        # call base class constructor
+        super().__init__(devname, properties=CurrInfoLinear._properties)
+
+    @property
+    def transpeff(self):
```

Comment:
I think you could add the charge readings from the ICTs. They are very useful.

Reply:
For LI, the ICTs readings are in ict device already. I could include the transport lines ICTs readings in ict device or merge the two devices in this currinfo (and add TB and TS ICTs readings). The transport efficiency are duplicated (in the currinfo IOC and in the ict IOC). I think it is better to use only the PVs provided by the currinfo IOC in the currinfo device and to add ICTs readings from TB and TS in the ict device (therefore related to icts IOCs only). What is your opinion?

Reply:
I think it would be useful to have the charges here too.. this way we don't have to create two devices everytime. You could add the ICT devices as subdevices of this one... But if you still feel uncomfortable with this option, no problem, you can leave it as is!

Reply:
@fernandohds564 please take I look in the two last commits.